### DISCUSSION TITLE
Stretching compute

### DISCUSSION THREAD

**Discussion by Baimanight** [ROBelgium/MSNoise#390] [General]
Hi, 
when I use the MSNoise1.6_compute_stretching, I noticed that the error values seem to be quite large. After reviewing the 'stretch.py' file, I suspect that there may have been an oversight in converting the error into the appropriate unit.  I made a modification that 'error = FWHM / 2 / nstr * str_range' for the error bars. 
Additionally, I've observed some null error values and instances where the errors exceed 'nstr' in the final station_pairs.csv file. Could you please help me understand why this might be happening? 
I appreciate your assistance and look forward to your helpful response.
![FWHM](https://github.com/user-attachments/assets/525996ff-7c97-4c77-95d3-9d42a5f73cdb)

**Comment by ThomasLecocq** [comment]
Hi, maybe @asyates or @LaureBrenot can help, I never used the stretching method extensively myself

**Comment by Baimanight** [comment]
Thanks for your response. Recently, based on your code, I am working with MATLAB to develop an improved stretching method for addressing clock error. And the final result looks good.


---Original---
From: "Thomas ***@***.***&gt;
Date: Mon, Jan 13, 2025 15:27 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ROBelgium/MSNoise] Stretching compute (Discussion #390)




 
Hi, maybe @asyates or @LaureBrenot can help, I never used the stretching method extensively myself

**Reply by ThomasLecocq** [reply]
Hi,good thanks. Please report here back with that code & suggestions on how to improve the processing, so others can benefit !

**Comment by Baimanight** [comment]
Sure, I’m just finishing up some more debugging. I’ll be happy to share and reply here again soon!






---Original---
From: "Thomas ***@***.***&gt;
Date: Mon, Jan 13, 2025 19:16 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ROBelgium/MSNoise] Stretching compute (Discussion #390)




 
Hi,good thanks. Please report here back with that code &amp; suggestions on how to improve the processing, so others can benefit !

**Comment by asyates** [comment]
Hi, 

Yes you're right, the error outputs were/are quite unusable in their current form (for the reason you describe, the lack of conversion into an appropriate unit). I've thus always used the max CC as an indicator of the uncertainty and ignored the error. 

I was thinking to update the error output to simply reflect the theoretical error i.e. Weaver et al. (2011), which is based on the max CC, frequency bandwidth, and coda window choices (after a bit of testing, i was not so convinced that the width of the CC estimates is that useful either, because this also depends on frequency). 

In this sense, would also be curious to see what you come up with before i go ahead with just introducing the theoretical error.

**Comment by ThomasLecocq** [comment]
@Baimanight , @asyates is currently working hard on msnoise, so if you have suggestions on the matter here, it'd be greatly appreciated!

**Comment by Baimanight** [comment]
@ThomasLecocq, @asyates After further investigation, I’ve drafted this [document](https://github.com/Baimanight/Stretching-optimization?tab=readme-ov-file) to outline the development process. However, I’m still new to the field of ambient noise, so there may be some obvious errors in my understanding. I would greatly appreciate your help in correcting any mistakes you notice.

Additionally, with the new semester approaching, I may have limited time to fully immerse myself in this project. As a result, this might be the first and final version for the time being.

**Reply by Baimanight** [reply]
By the way, I noticed the error related to FWHM doesn’t seem impactful, so I’ve chosen to disregard it for now

**Reply by asyates** [reply]
@Baimanight impressive work! i will take a closer look shortly and get back to you on any specifics. I'm still planning to re-visit the stretching function in the near future to make various changes, including the error output (as you say, the FWHM isn't so useful, at least in its present form). Looks like your work provides some nice things to think about.
